File: labs/lesson-001/list_models.rb
require 'aws-sdk-bedrock'

begin
  puts "Initializing Amazon Bedrock control-plane client..."
  # Initializes using default AWS credential provider chain
  client = Aws::Bedrock::Client.new(region: 'us-east-1')

  puts "Fetching available foundation models..."
  response = client.list_foundation_models

  puts "\n%-40s | %-30s | %-15s" % ["MODEL NAME", "MODEL ID", "PROVIDER"]
  puts "-" * 90

  response.model_summaries.take(15).each do |model|
    puts "%-40s | %-30s | %-15s" % [model.model_name, model.model_id, model.provider_name]
  end

rescue LoadError
  puts "Error: The 'aws-sdk-bedrock' gem is not installed."
  puts "Please run: gem install aws-sdk-bedrock"
rescue Aws::Bedrock::Errors::ServiceError => e
  puts "AWS Service Error: #{e.message}"
rescue StandardError => e
  puts "An unexpected error occurred: #{e.message}"
end

File: labs/lesson-001/list_models.js
const { BedrockClient, ListFoundationModelsCommand } = require("@aws-sdk/client-bedrock");

async function listModels() {
  console.log("Initializing Amazon Bedrock control-plane client...");
  const client = new BedrockClient({ region: "us-east-1" });

  try {
    const command = new ListFoundationModelsCommand({});
    const response = await client.send(command);

    console.log("\nAvailable Foundation Models (First 15):");
    console.log("=".repeat(80));
    
    const limit = response.modelSummaries.slice(0, 15);
    limit.forEach(model => {
      console.log(`- ${model.modelName} (ID: ${model.modelId}) [Provider: ${model.providerName}]`);
    });

  } catch (error) {
    if (error.code === "MODULE_NOT_FOUND") {
      console.log("Error: @aws-sdk/client-bedrock is not installed.");
      console.log("Please run: npm install @aws-sdk/client-bedrock");
    } else {
      console.error("AWS API Error:", error.message || error);
    }
  }
}

listModels();
